index.js

```diff
@@ -1,6 +1,73 @@
-
 import { NativeModules } from 'react-native';
 
 const { RNSnowplowTracker } = NativeModules;
 
-export default RNSnowplowTracker;
+export default class Tracker {
+
+  static initialize(argmap) {
+
+    let defaults = {// defaults for optional params
+                 method: 'post',
+                 protocol: 'https',
+                 setPlatformContext : false,
+                 base64Encoded : false,
+                 platformContext : false,
+                 lifecycleEvents : false,
+                 screenContext : false,
+                 sessionContext : false,
+                 foregroundTimeout : 600,
+                 backgroundTimeout : 300,
+                 checkInterval : 15,
+                 installTracking : false}
+
+    if (typeof argmap.endpoint !== 'undefined' && typeof argmap.appId !== 'undefined' && typeof argmap.namespace !== 'undefined' ) {
+      return RNSnowplowTracker.initialize({...defaults, ...argmap});
```

Comment:
Potentially a dev experience improvement here would be to manage or hold an instance of this `initPromise` internally as well so when `trackScreenViewEvent` was called you could await it in that function internally rather than the scenario where a user forgets to await it outside and calls a track method before the promise resolves and it causes issues.

Reply:
Ty Elliot. Shame not to have time to do this in this PR, but I've logged [an issue](https://github.com/snowplow-incubator/snowplow-react-native-tracker/issues/78) for next release!